src/main/scala/rps/Main.scala

```diff
@@ -1,35 +1,69 @@
 package rps
-import scala.io.StdIn.readLine
 import scala.util.Random
 import io.buildo.enumero.annotations.enum
 import io.buildo.enumero.{CaseEnumIndex, CaseEnumSerialization}
 import Move._
+import Response._
 import Result._
-import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
 import io.circe.generic.auto._
 import io.buildo.enumero.circe._
 import akka.actor.ActorSystem
-import akka.http.scaladsl.Http
-import akka.http.scaladsl.model._
-import akka.http.scaladsl.server.Directives._
 import akka.stream.ActorMaterializer
-import scala.io.StdIn
+import wiro.annotation._
+import scala.concurrent.{ExecutionContext, Future}
+import wiro.Config
+import wiro.server.akkaHttp._
+import wiro.server.akkaHttp.FailSupport._
 
-object Main extends App {
-  WebServer.run
+@enum trait Move {
+  object Rock
+  object Paper
+  object Scissors
+}
+
+@enum trait Result {
+  object Win
+  object Lose
+  object Draw
+}
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import wiro.Config
+import wiro.server.akkaHttp._
+
+object WebServer extends App with RouterDerivationModule {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  implicit def throwableResponse: ToHttpResponse[Throwable] = null
```

Comment:
`null` should never be used in Scala! if you don't want to implement a method, just use `???` that will throw a `NotImplementedException`.

But you could implement the error handling (even if in your app you never return a Left, in real-world application you must handle them! here you define what kind of response/statuscode returning to the client, for example:
 
```suggestion
import akka.http.scaladsl.model.{ContentType, HttpEntity, HttpResponse, MediaTypes, StatusCodes}
import io.circe.syntax._

 implicit def throwableResponse: ToHttpResponse[Throwable] = error => 
 HttpResponse(
 status = StatusCodes.InternalServerError,
 entity = HttpEntity(
 ContentType(MediaTypes.`application/json`),
 error.getMessage().asJson.noSpaces))
```

in a real-world application, you could have a custom Error instead of generic Throwable, and you could pattern-match on the error to find the best code to return